lib/mwoffliner.lib.js

```diff
@@ -911,6 +927,99 @@ module.exports = {
                 }
             }
 
+            function storeDependencies( parsoidDoc, articleId, finished ) {
```

Comment:
This method - at least all the part which does not directly impact the footer of the article, should run only one time for each ZIM. This would save us a lot of file system access.